### PR TITLE
Adds option to replace time to deal with delayed metrics.

### PR DESCRIPTION
**Pull request by nicholaskuechler**

### CODE REVIEW

**Comment by VinnyQ** [comment]
@nicholaskuechler not sure why the tests are failing, can you try running it locally and see if it passes?  could be our travis setup.

File: twisted/plugins/graphite_blueflood_plugin.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Options(usage.Options):
         ['key',           'k', '', 'Rackspace authentication password. It is recommended not to set this option from the command line, as that can compromise api keys. Instead, set the key in a config file and use the \'--config\' option below.'],
         ['auth_url',      '',  AUTH_URL, 'Auth URL'],
         ['limit',         '',  0, 'Blueflood json payload limit, bytes. 0 means no limit'],
+        ['replace_time'   '',  False, 'Replace metric time with current blueflood carbon forwarder node time'],
```
Comment (VinnyQ) [inline twisted/plugins/graphite_blueflood_plugin.py]: missing a , here after 'replace_time'

File: bluefloodserver/blueflood.py [hunk 2]
```diff
@@ -46,22 +54,26 @@ def __init__(self, ingest_url='http://localhost:19000', retrieve_url='http://loc
         self.headers = {}
         self.limit = limit
         self._buffer_size = 0
+        self.replace_time = replace_time
+
+    def ingest(self, metric_name, metric_time, value, ttl):
+        if self.replace_time:
+            metric_time = time.time()
```
Comment (VinnyQ) [inline bluefloodserver/blueflood.py]: let's also set this as a list with `[time.time()]`

**Comment by VinnyQ** [comment]
@nicholaskuechler keekz make the two changes I commented in the code and let's see if it'll fix the tests.  Let me know if you are unclear what the comments mean.

File: example.conf [hunk 3]
```diff
@@ -10,3 +10,4 @@ user=
 key=
 auth_url=https://identity.api.rackspacecloud.com/v2.0/tokens
 limit=0
+replace_time=False
```
Comment (VinnyQ) [inline example.conf]: Not sure if I like the name `replace_time` but if everyone is ok with it I am too.

I would mention it's the collection time that we're replacing with now() somehow, maybe `replace_metric_time` or even `replace_metric_time_on_flush`

Comment (shintasmith) [inline example.conf]: How about ```overwrite_collection_time``` ?

**Comment by shintasmith** [comment]
Don't forget to also update the README with the new optional config

File: example.conf (hunk 3 above)
Comment (VinnyQ) [inline example.conf]: +1 'overwrite_collection_time'

Let's make that change, @nicholaskuechler

**Comment by VinnyQ** [comment]
👍 LGTM!  Let's get one more +1 and we can merge.  Maybe have @usnavi look at this to make sure he's ok with it.

File: README.md [hunk 4]
```diff
@@ -38,6 +38,7 @@ It does very primitive _'caching'_: aggregates all metrics and flushes them in r
 | -k | Keystone key | |
 | --auth-url | Keystone token URL | |
 | --config | Set options from a config file | |
+| --overwrite_collection_timestamp | Replace metric collection timestamp with ingest timestamp | False |
```
Comment (usnavi) [inline README.md]: Can this be passed in through the config file?  Does https://github.com/rackerlabs/blueflood-carbon-forwarder/blob/9270aee430ea59cad0a0ae78a7bc78b7c357637d/twisted/plugins/graphite_blueflood_plugin.py#L21 need to be updated for this?

**Comment by nicholaskuechler** [comment]
@usnavi I think you're looking at the original and not the diff. My diff includes that: https://github.com/rackerlabs/blueflood-carbon-forwarder/pull/18/files#diff-220c2e413d1e1469bd26294d376aa1b8R35

**Pull request merged**